obf-generation/travel-gpx/README.md

```diff
@@ -0,0 +1,144 @@
+GPX=osmand:line_3d_visualization_position_type OBF=gpx_line_3d_visualization_position_type POI,FLEXIBLE
+GPX=osmand:line_3d_visualization_wall_color_type OBF=gpx_line_3d_visualization_wall_color_type POI,FLEXIBLE
+GPX=osmand:show_arrows OBF=gpx_show_arrows POI,FLEXIBLE
+GPX=osmand:show_start_finish OBF=gpx_show_start_finish POI,FLEXIBLE
+GPX=osmand:split_interval OBF=gpx_split_interval POI,FLEXIBLE
+GPX=osmand:split_type OBF=gpx_split_type POI,FLEXIBLE
+GPX=osmand:vertical_exaggeration_scale OBF=gpx_vertical_exaggeration_scale POI,FLEXIBLE
+```
+
+### 2.2. Waypoints
+
+```
+GPX=osmand:color OBF=color POI,MAP,STYLE,SPECIAL
+GPX=osmand:icon OBF=gpx_icon POI,STYLE
+GPX=osmand:background OBF=gpx_bg POI,STYLE
+GPX=<type> OBF=points_groups_category POI
+```
+
+## 3. Save/Restore (shields and OSM-related tags)
+
+```
+GPX=osmand:osm_tag_* OBF=[POI|MAP] POI,MAP,STYLE,FLEXIBLE (some OSM tags will be lost: osmc:symbol, route, type, etc)
+
+GPX=osmand:use_osmc_colors OBF=use_osmc_colors POI,MAP,STYLE
+
+GPX=osmand:osm_id OBF=osm_id POI,MAP (will be used to keep/display OSM id if applicable)
+GPX=osmand:osm_ref_present OBF=gpx_osm_ref_present POI ("yes" if "ref" comes from osm tags)
+GPX=osmand:osm_name_present OBF=gpx_osm_name_present POI ("yes" if "name" comes from osm tags)
+GPX=osmand:osm_description_present OBF=gpx_osm_description_present POI ("yes" if "description" comes from osm tags)
+
+GPX=osmand:shield_bg OBF=shield_bg POI,MAP,STYLE
+GPX=osmand:shield_fg OBF=shield_fg POI,MAP,STYLE
+GPX=osmand:shield_fg_2 OBF=shield_fg_2 POI,MAP,STYLE
+GPX=osmand:shield_text OBF=shield_text POI,MAP,STYLE
+GPX=osmand:shield_textcolor OBF=shield_textcolor POI,MAP,STYLE
+GPX=osmand:shield_waycolor OBF=shield_waycolor POI,SPECIAL (shadow copy of osmc_waycolor)
+```
+
+## 4. Save/Restore (analytics, elevation and special tags)
+
+```
+OBF=route_id POI,MAP,SPECIAL (used to identify unique GPX and to connect Map<->POI data)
+
+OBF=route_radius POI (specify default radius to search in POI section)
+
+OBF=distance POI (distance of all GPX segments)
+OBF=max_speed,avg_speed,min_speed POI (speed analytics)
+OBF=time_span,time_span_no_gaps,time_moving,time_moving_no_gaps POI (time analytics)
+OBF=start_ele,ele_graph MAP (GPX-specified elevation data wrapped in compact binary array)
+OBF=min_ele,avg_ele,max_ele,diff_ele_up,diff_ele_down POI,MAP (analytics of GPX-specified elevation data)
+```
+
+## 5. Rendering (Map-section)
+
+### 5.1. Tracks
+
+```
+GPX=<trkseg> OBF=route VAL=segment
+
+GPX=osmand:color OBF=color POI,MAP,STYLE,SPECIAL
+GPX=osmand:width OBF=gpx_width POI,MAP,STYLE (fixed width values supported)
+
+GPX=osmand:shield_bg OBF=shield_bg POI,MAP,STYLE
+GPX=osmand:shield_fg OBF=shield_fg POI,MAP,STYLE
+GPX=osmand:shield_fg_2 OBF=shield_fg_2 POI,MAP,STYLE
+GPX=osmand:shield_text OBF=shield_text POI,MAP,STYLE
+GPX=osmand:shield_textcolor OBF=shield_textcolor POI,MAP,STYLE
+
+GPX=osmand:use_osmc_colors OBF=use_osmc_colors POI,MAP,STYLE (transparent colors)
```

Comment:
why transparent colors are not encoded as transparent into color? If color is a fixed set of colors, than tag should be renamed to osmand:transparent_lines=true ?

Reply:
`use_osmc_colors` defines the translation of the fixed colors set to semi-transparent colors, partially defined by $osmc*Color prefixes, and defines flexible **width** based on zoom (not based on gpx_width tag)

use_osmc_colors, transparent_lines, *transparent* all are not ok

What about split into 2 tags:

`flexible_line_width=yes`
`translucent_line_colors=yes`

PS current xml:

```
<switch minzoom="11" tag="route" value="segment" additional="use_osmc_colors=yes" color="$osmcDefaultColor" cap="ROUND">
  <apply_if additional="color_white" color="#44ffffff"/>
  <apply_if additional="color_gray" color="#44999999"/>
  <apply_if additional="color_darkyellow" color="#44afb61c"/>
  <apply_if additional="color_lightgreen" color="#446cb52b"/>
  <apply_if additional="color_lightblue" color="#442ec6ff"/>
  <apply_if additional="color_red" color="$osmcRedColor"/>
  <apply_if additional="color_orange" color="$osmcOrangeColor"/>
  <apply_if additional="color_brown" color="$osmcBrownColor"/>
  <apply_if additional="color_yellow" color="$osmcYellowColor"/>
  <apply_if additional="color_green" color="$osmcGreenColor"/>
  <apply_if additional="color_blue" color="$osmcBlueColor"/>
  <apply_if additional="color_purple" color="$osmcPurpleColor"/>
  <apply_if additional="color_black" color="$osmcBlackColor"/>
  <case maxzoom="11" strokeWidth="1"/>
  <case maxzoom="12" strokeWidth="2"/>
  <case maxzoom="13" strokeWidth="3"/>
  <case maxzoom="14" strokeWidth="4"/>
  <case maxzoom="15" strokeWidth="4"/>
  <case maxzoom="16" strokeWidth="5"/>
  <case maxzoom="17" strokeWidth="6"/>
  <case minzoom="18" strokeWidth="7"/>
</switch>
```

Reply:
refactored into 2 understandable tags